sigstore/_internal/fulcio/client.py

```diff
@@ -264,35 +148,12 @@ def post(
         cert = load_pem_x509_certificate(certificates[0].encode())
         chain = [load_pem_x509_certificate(c.encode()) for c in certificates[1:]]
 
-        if sct_embedded:
-            try:
-                # The SignedCertificateTimestamp should be acessed by the index 0
-                sct = _get_precertificate_signed_certificate_timestamps(cert)[0]
-
-            except UnexpectedSctCountException as ex:
-                raise FulcioClientError(ex)
-
-        else:
-            # If we don't have any embedded SCTs, then we might be dealing
-            # with a Fulcio instance that provides detached SCTs.
-
-            # The detached SCT is a base64-encoded payload, which in turn
-            # is a JSON representation of the SignedCertificateTimestamp
-            # in RFC 6962 (subsec. 3.2).
-            try:
-                sct_b64 = resp.json()["signedCertificateDetachedSct"][
-                    "signedCertificateTimestamp"
-                ]
-            except KeyError:
-                raise FulcioClientError(
-                    "Fulcio response did not include a detached SCT"
-                )
-
-            try:
-                sct = DetachedFulcioSCT.model_validate_json(base64.b64decode(sct_b64))
-            except Exception as exc:
-                # Ideally we'd catch something less generic here.
-                raise FulcioClientError from exc
+        try:
+            # The SignedCertificateTimestamp should be accessed by the index 0
+            sct = _get_precertificate_signed_certificate_timestamps(cert)[0]
+
+        except UnexpectedSctCountException as ex:
+            raise FulcioClientError(ex)
```

Comment:
itching to refactor these methods a bit, using them is overly complicated and it's still incomplete (you could get  ValueError in addition to this unnecessary internal error)... but that's not really related to this PR

Reply:
Yeah, I'd love to have these cleaned up. Let's do a follow-up for that 🙂